src/Microsoft.AspNetCore.Blazor.Razor.Extensions/BlazorDiagnosticFactory.cs

```diff
@@ -308,6 +308,16 @@ public static RazorDiagnostic Create_ChildContentHasInvalidParameter(SourceSpan?
             var attributesText = string.Join(", ", attributes.Select(a => $"'{a.Name}'"));
             return RazorDiagnostic.Create(GenericComponentTypeInferenceUnderspecified, source ?? SourceSpan.Undefined, component.TagName, attributesText);
         }
-    }
 
+        public static readonly RazorDiagnosticDescriptor ChildContentHasInvalidParameterOnComponent =
+            new RazorDiagnosticDescriptor(
```

Comment:
Added a new error for component-scoped invalid parameters. The existing error message would be confusing as it mentions child content elements specifically.